src/smccc.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Decoder for SMC Calling Convention ARM DEN 0028E v1.4
+
+mod smccc_arm;
+mod smccc_common;
+mod smccc_hyp;
+mod smccc_secure;
+mod smccc_tapp;
+
+use smccc_arm::decode_arm_service;
+use smccc_common::decode_common_service;
+use smccc_common::describe_general32_queries;
+use smccc_common::reserved_fids;
+use smccc_common::smccc_general32_queries;
+use smccc_hyp::decode_hyp_service;
+use smccc_secure::decode_secure_service;
+use smccc_tapp::decode_tapp_service;
+
+use super::{DecodeError, FieldInfo};
+
+pub fn decode_smccc(smccc: u64) -> Result<Vec<FieldInfo>, DecodeError> {
```

Comment:
It would be good to add a Rustdoc comment for this function, as it's part of the public API and will show up at https://docs.rs/aarch64-esr-decoder/latest/aarch64_esr_decoder/.